src/client/Client.cc

```diff
@@ -6135,7 +6135,7 @@ int Client::_lookup(Inode *dir, const string& dname, int mask, InodeRef *target,
 	     << " seq " << dn->lease_seq
 	     << dendl;
 
-    if (!dn->inode || dn->inode->caps_issued_mask(mask)) {
+    if (!dn->inode || dn->inode->caps_issued_mask(mask, true)) {
```

Comment:
I'm a bit leery of doing this in _lookup() because of how it's invoked from Client::path_walk(). Although looking at it more carefully it seems okay, so maybe I'm just having flashbacks to dealing with the MDS' more complicated version.

Did anybody else check out those interactions?

Reply:
I think it's ok in this lookup case.

In the event that caps_issued_mask returns true there, we just end up satisfying the request out of the cache. If one of the other checks fails and we have to call out to the MDS anyway, then no harm done.

Do let me know if you see potential problems here though.

Reply:
Well, if we have to call out to the MDS anyway, that doesn't imply we restart the lookup. We just issue an MDS request based on our current location, then keep going.

But I guess that isn't actually any different than if we had to call out to the MDS and then got a cap revoke on one of the ancestors farther up the tree, so this should be okay.

Reply:
Yes, we can't guarantee the integrity of the entire path over a pathwalk. Note that the kernel doesn't do that either, fwiw, so this should be ok.